packages/path_provider/path_provider_platform_interface/lib/src/method_channel_path_provider.dart

```diff
@@ -78,8 +78,9 @@ class MethodChannelPathProvider extends PathProviderPlatform {
   }
 
   Future<String> getDownloadsPath() {
-    if (!_platform.isMacOS) {
-      throw UnsupportedError('Functionality only available on macOS');
+    if (!_platform.isMacOS && !_platform.isWindows) {
```

Comment:
This should not be opt-in; it undermines one of the goals of federated plugins. It also doesn't address the fact that Linux is incorrectly blocked here, and fixes for this more general problem shouldn't be ad-hoc.

I don't know why this plugin does pre-screening by platform at this layer in the first place, so someone familiar with the history of this pattern in path_provider will need to review this. If maintainers of the plugin want to keep a check, it should be an opt-out (e.g., return an error for iOS and Android), not opt-in, but my preference would be to follow the more normal plugin pattern of simply allowing the implementation to handle—or not handle—the message.

Reply:
I removed the pre-screening by platform as suggested by @stuartmorgan .